shared/test/dataDriver_test.js

```diff
@@ -151,6 +154,34 @@ describe("loop.DataDriver", () => {
     });
   });
 
+  describe("#handleEvent.error", () => {
+    it("should retry the connection on error", () => {
+        // This calls _connectToRoom, so don't stub it out until after.
+        driver.setupWindowData(new actions.SetupWindowData({
+          roomToken: "fakeSetup"
+        }));
+        sandbox.stub(driver, "_connectToRoom");
+        driver.handleEvent({ type: "error" });
+        clock.tick(driver.INITIAL_RETRY_TIMEOUT + 1);
+        sinon.assert.called(driver._connectToRoom);
+    });
+
+    it("should exponentially back off while retrying", () => {
+        driver.setupWindowData(new actions.SetupWindowData({
+          roomToken: "fakeSetup"
+        }));
+        driver.handleEvent({ type: "error" });
+        clock.tick(driver.INITIAL_RETRY_TIMEOUT + 1);
```

Comment:
I'm not sure if we actually want to tick the clock. And it's not stubbing `_connectToRoom` so it'll actually try to connect to room 3 times. You could consolidate the `SetupWindowData` and stubbing to a `beforeEach`